### PR TITLE
Setup renovate for updating dependencies

### PR DESCRIPTION
**Pull request by AlexWaygood**
This PR sets up a [renovate](https://docs.renovatebot.com/) configuration file for updating dependencies. It is a competing PR to #11564.

The proposed configuration:
- Checks daily for new versions of pyright, pytype, and flake8-pyi
- Checks quarterly for new versions of all other pip dependencies (except for mypy), and groups them into a single PR
- Checks quarterly for new versions of all GitHub Actions, and groups them into a single PR

The advantages of renovate over dependabot are:
- It's extremely configurable. That means we can keep our pyright pin in `pyproject.toml` by defining a [custom "manager"](https://docs.renovatebot.com/modules/manager/regex/) that uses a regex to grab the pyright version from our `pyproject.toml` file and check to see if it needs updates.
- We can set a quarterly schedule for GitHub Actions rather than a monthly schedule

The disadvantages of renovate over dependabot are:
- It's extremely configurable. That means it's easier to get the configuration wrong, and there's more complexity to wade through when navigating their docs.
- We'd need to [authorise renovate's Github.com app](https://docs.renovatebot.com/getting-started/installing-onboarding/#hosted-githubcom-app) to have access to this repository. Possibly that might require an admin for the `python` organisation rather than just an admin for typeshed? (I am neither, so don't really know.)

Fixes https://github.com/python/typeshed/issues/11484
Closes https://github.com/python/typeshed/pull/11491
Closes #11564

### CODE REVIEW

**Comment by AlexWaygood** [comment]
I use renovate over at typeshed-stats, FWIW, and I've been reasonably happy with it. Here's an example PR:
- https://github.com/AlexWaygood/typeshed-stats/pull/181

**Comment by jakebailey** [comment]
> * We'd need to [authorise renovate's Github.com app](https://docs.renovatebot.com/getting-started/installing-onboarding/#hosted-githubcom-app) to have access to this repository. Possibly that might require an admin for the `python` organisation rather than just an admin for typeshed? (I am neither, so don't really know.)

IIRC someone does need to add it to the org, but the app can be installed with only perms for one repo.

Maybe one day dependabot will get this... https://github.com/dependabot/dependabot-core/issues/2483#issuecomment-1212433459

**Comment by JelleZijlstra** [comment]
I pushed some buttons but it doesn't seem like I have access to authorize the app here either; we'll probably have to ping Łukasz to do it.

File: .github/renovate.json [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+        "matchManagers": ["github-actions"],
+        "description": "Group all GitHub Action dependencies",
+        "separateMajorMinor": "false",
+        "schedule": ["every 3 months on the first day of the month"]
+      },
+      {
+        "groupName": "Monthly pip dependencies",
+        "matchManagers": ["pip_requirements"],
+        "excludePackageNames": ["mypy", "pytype", "flake8-pyi"],
+        "description": "Monthly update of most pip dependencies",
```
Comment (JelleZijlstra) [inline .github/renovate.json:27]: Monthly or 3-monthly?

**Comment by AlexWaygood** [comment]
> I pushed some buttons but it doesn't seem like I have access to authorize the app here either; we'll probably have to ping Łukasz to do it.

Well, let's decide amongst ourselves whether this is a solution we like before we ping him 😄

File: .github/renovate.json (hunk 1 above)
Comment (AlexWaygood) [reply .github/renovate.json:27]: Good catch!

**Comment by AlexWaygood** [comment]
I looked into switching off pre-commit autoupdate PRs altogether, but it seems that's not possible (https://github.com/pre-commit-ci/issues/issues/83). So it's possible we'll get PRs from both renovate and pre-commit on the first day of each quarter, if we go with this. I think we can live with that, though, personally.

**Pull request merged**